### PR TITLE
Raffle Entry Page

### LINKED ISSUE #4: Raffle Entry Page for Participants
We need to create an entry page for raffle where anyone can come in, enter the ID of raffle and be redirected to the event page.

### PR DESCRIPTION
**Pull request by ErtyumPX**
Closes #4 
Gently takes forward #5 
Touches the first and the weakest protection way in #6 

## What are we doing in this PR?

This PR mainly focuses on raffle entry page, which is also the main landing page of the website.

- Adds `react-spinners` package for loading screens
- Creates a easily reusable `<LoadingScreen/>` component that fades the background and has a spinning clock in the middle
- Creates a pseudo api function that will be used later to check if the entered raffle id is valid or not.
- Creates a `LocalService` for general local storage interactions. Checks if the raffle id is in the submitted ids list.
- IP check will also be added later when the backend is ready.
- InputBox in the page is a `<NumberInput/>` by Tremor, it does not allowany character but numbers by default.

## How does the UI look like?

Here is a picture of the Raffle Entry page in dark and light mode:

screenshot-2024-03-27_11:44:06

screenshot-2024-03-27_11:44:16

## What will be done as follow-up?

We will be working on the raffle submission page, to get the raffle information and display the questions, necessary raffle descriptions etc.

When backend is ready, IP check will be added to see if someone was participated using the current IP address of the device.

## Notes

To make it have a sense, we also need to check local storage submitted ids in raffle submission page.

It could have been much nicer to center the text in the `<NumberInput/>` but I could not figure out how to do so.

**Do not forget about `yarn install` before your morning water!**

### CODE REVIEW

**Comment by ErtyumPX** [comment]
@Utkub24 @HuseyinSimsek7904 when do you guys think you can check the PRs?

File: src/app/_app.ts
Comment (HuseyinSimsek7904) [inline src/app/_app.ts:1]: Whats this file?

**Review by HuseyinSimsek7904 — APPROVED** [review]
Since I am not heavily working on this repository, I am not sure I understand everything perfectly. However, the changes seem to be alright and good work!

File: src/app/_app.ts
Comment (ErtyumPX) [reply src/app/_app.ts:1]: It's mostly used for configuring the app directly, I accidentally added and forgot to remove it, I believe I'm removing it in the follow-up PRs, just ignore it :+1:

**Review by Utkub24 — APPROVED** [review]
Cool beans!

P.S. Is there way of disabling the text cursor from showing up on the text/buttons?
cursor1
cursor2

**Comment by Utkub24** [comment]
> @Utkub24 I think thats the caret browsing feature in firefox, press F7 to turn that off.
Huh TIL!

**Pull request merged**